exercise7/index.js

```diff
@@ -0,0 +1,49 @@
+// This code will run when the page loads.
+window.addEventListener("load", () => {
+  // Get all the elements on the page that have click actions or will be changed as the game is played.
+  // For starters, I've added the message div and the reset button.
+  let rows = document.querySelectorAll(".row");
+  let listEntries = document.querySelectorAll(".listEntry");
+  let checkboxes = document.querySelectorAll("input[name=checkbox]");
+  const submitButton = document.getElementById("submitNew");
+  const addBox = document.getElementById("addNew");
+  const container = document.querySelector(".container");
+
+  submitButton.addEventListener("click", function () {
+    let submitText = addBox.value;
+
+    let row = document.createElement("div");
+    row.setAttribute("class", "row");
+    row.setAttribute("id", "r" + rows.length);
+    container.appendChild(row);
+    rows = document.querySelectorAll(".row");
+
+    let list = document.createElement("div");
+    list.setAttribute("class", "listEntry");
+    list.setAttribute("id", "l" + listEntries.length);
+    list.innerHTML = submitText;
+    rows[listEntries.length].appendChild(list);
+    listEntries = document.querySelectorAll(".listEntry");
+
+    let cb = document.createElement("input");
+    cb.setAttribute("type", "checkbox");
+    cb.setAttribute("name", "checkbox");
+    cb.setAttribute("id", "c" + checkboxes.length);
+    listEntries[checkboxes.length].appendChild(cb);
+    checkboxes = document.querySelectorAll("input[name=checkbox]");
+
+    for (let n = 0; n < checkboxes.length; n++) {
```

Comment:
This is adding a change listener to all the checkboxes, not just the newly added one. That will cause more and more change listeners to be added to the items already on the list as new items are added. You should only add listeners to the new elements.

Also, the way you're taking listEntries[n] also causes a bug because since the list is reordered, it won't always refer to the same item. Follow these steps on your list to see the bug:

1. Add "Item 1"
2. Add "Item 2"
3. Cross off "Item 1"
4. Add "Item 3"
5. Uncross "Item 1"
6. Cross off "Item 2"

Reply:
Updating listener code

Reply:
Updated list sorting code